### ISSUE TITLE
Exposè darwin-rebuild as a Flake package/app

### ISSUE THREAD

**Issue by hauleth** [LnL7/nix-darwin#613]
This will allow making the Flakes more self contained and manageable as it would allow user to define their own application that would allow calling setup via `nix run`.

**Comment by srid** [comment]
FTR, this is how I do it: https://github.com/srid/nixos-flake/blob/fab75907cfc0c4e48ceab7a7194f019b55828895/flake-module.nix#L54-L65

**Comment by hauleth** [comment]
@srid thanks, using your approach I have prepared mine, slightly different:
```nix
      apps = {
        system = let
          emptyConfiguration = darwin.lib.darwinSystem {
            inherit system;

            modules = [];
          };
          builder =
            if pkgs.stdenv.isDarwin
            then "${emptyConfiguration.system}/sw/bin/darwin-rebuild"
            else pkgs.lib.getExe pkgs.nixos-rebuid;
        in {
          type = "app";

          program = toString (pkgs.writeScript "activate-system" ''
            ${builder} --flake "${self}#''$(hostname -s)" "$@"
          '');
        };
      };
```

**Comment by hauleth** [comment]
@garrettmichaelgeorge I did that on purpose ;) This allows me to do `nix run my-flake#system -- switch` to make a switch, but it also allows me to run it with any other subcommand I need.

**Comment by srid** [comment]
We could make `switch` the default unless the user has passed custom args. https://github.com/srid/proc-flake/blob/676b5af467e52c06aa8a9165edb3345c6dbbabde/flake-module.nix#L56-L58

**Comment by mirkolenz** [comment]
This is what I did in my config:

```nix
let
  builder =
    if pkgs.stdenv.isDarwin
    then
      let
        emptyDarwin = darwin.lib.darwinSystem {
          inherit system;
          modules = [ ];
        };
      in
      "${emptyDarwin.system}/sw/bin/darwin-rebuild"
    else pkgs.lib.getExe pkgs.nixos-rebuild;
in
{
  type = "app";
  program = builtins.toString (pkgs.writeShellScript "rebuild" ''
    set -x #echo on
    REBUILD_TYPE=''${1:-switch}
    ${builder} --flake ${self} "$REBUILD_TYPE" "''${@:2}"
  '');
}
```

I also omitted the `hostname` subcommand since that is the default behavior anyway.

**Comment by Enzime** [comment]
Fixed by #687, `darwin-rebuild` is the default package so you can run it with `nix run nix-darwin` :smiley:
